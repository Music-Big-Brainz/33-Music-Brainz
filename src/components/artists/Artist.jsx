import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Artist = ({ artistId, name, disambiguation }) => (
  <Link to={`/releases/${artistId}`}>
    <figure>
      <p>{name}</p>
      <p>{disambiguation}</p>
    </figure>
  </Link>
);

Artist.propTypes = {
  artistId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  disambiguation: PropTypes.string.isRequired,
};

export default Artist;
