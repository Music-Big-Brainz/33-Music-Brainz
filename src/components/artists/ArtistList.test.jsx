import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Artist from './Artist';
import { MemoryRouter } from 'react-router-dom';

describe('displays artists', () => {
  afterEach(() => cleanup());
  it('renders artists', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Artist artists="" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
