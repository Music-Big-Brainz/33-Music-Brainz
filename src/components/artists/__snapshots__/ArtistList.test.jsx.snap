// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays artists renders artists 1`] = `
<DocumentFragment>
  <a
    href="/releases/Prince/070d193a-845c-479f-980e-bef15710653e"
  >
    <figure>
      <p>
        Prince
      </p>
      <p>
        the artist formely known as
      </p>
    </figure>
  </a>
</DocumentFragment>
`;
